python/paddle/fluid/incubate/fleet/base/role_maker.py

```diff
@@ -239,3 +246,32 @@ def _generate_role(self):
                 self.node_type_ = 1
             self.node_type_comm_ = self.comm_.Split(self.node_type_)
             self.role_is_generated_ = True
+
+
+class UserDefinedRoleMaker(RoleMakerBase):
```

Comment:
I think you can add a local role maker that can be easily configured. For example, the ip is 127.0.0.0 by default and port id is random generated within a range.

Reply:
yep, I will seriously consider.